magda-ckan-connector/src/Ckan.ts

```diff
@@ -147,29 +154,34 @@ export default class Ckan implements ConnectorSource {
         });
     }
 
-    public getJsonDatasets(): AsyncPage<any[]> {
+    public getJsonDatasets(): AsyncPage<DatasetContainer[]> {
+        
         const packagePages = this.packageSearch({
             ignoreHarvestSources: this.ignoreHarvestSources,
             sort: 'metadata_created asc'
         });
-        return packagePages.map((packagePage) => packagePage.result.results);
+    
+        return packagePages.map((packagePage) => packagePage.result.results.map(ckanDataset => new DatasetContainer(ckanDataset, packagePage.retrievedAt)));
     }
 
-    public getJsonDataset(id: string): Promise<any> {
+    public getJsonDataset(id: string): Promise<(DatasetContainer)> {
         const url = this.urlBuilder.getPackageShowUrl(id);
 
-        return new Promise<any>((resolve, reject) => {
+        return new Promise<(DatasetContainer)>((resolve, reject) => {
             request(url, { json: true }, (error, response, body) => {
                 if (error) {
                     reject(error);
                     return;
                 }
+                //TODO ask Kevin if this is appropriate
```

Comment:
It would be better to make the retrievedAt date the date when the request started, rather than when it finished.  In other words, call `moment()` before the call to `request(...)`.